src/BaseFactory/BaseFactory.sol

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+pragma solidity ^0.8.0;
+
+import {IFactory} from "./interfaces/IFactory.sol";
+
+/// @title BaseFactory
+/// @author Euler Labs (https://www.eulerlabs.com/)
+/// @notice A minimal factory for deploying various contracts.
+abstract contract BaseFactory is IFactory {
+    /// @notice Contracts deployed by the factory.
+    mapping(address => DeploymentInfo) internal deploymentInfo;
+
+    /// @notice An array of addresses of all the contracts deployed by the factory
+    address[] public deployments;
+
+    /// @inheritdoc IFactory
+    function getDeploymentInfo(address contractAddress) external view returns (address deployer, uint96 deployedAt) {
```

Comment:
I actually don't remember why I made `deploymentInfo` internal and added an external getter method to deconstruct `DeploymentInfo`. 

We could just make `deploymentInfo` public and remove this getter.

Reply:
True. On the other hand having the function declared in the interface has some benefits too. I'd keep it.